From 7b35a82d424b403c80894247e0e1107bb3d4a1f6 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Tue, 3 Mar 2026 14:49:33 +0100
Subject: [PATCH] COMP: Install .txx template files in MRMLCore development
 headers

vtkMRMLCoreTestingUtilities.h includes vtkMRMLCoreTestingUtilities.txx
via a relative #include. The MRMLCore install glob only captured *.h,
leaving the .txx file uninstalled. Extension test builds that include
vtkMRMLCoreTestingMacros.h (which pulls in vtkMRMLCoreTestingUtilities.h)
then fail with "No such file or directory" for the .txx file.
---
 Libs/MRML/Core/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Libs/MRML/Core/CMakeLists.txt b/Libs/MRML/Core/CMakeLists.txt
index f8a18bf425..63b4e668d9 100644
--- a/Libs/MRML/Core/CMakeLists.txt
+++ b/Libs/MRML/Core/CMakeLists.txt
@@ -129,7 +129,8 @@ if(NOT DEFINED ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   set(${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT ON)
 endif()
 if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
-  file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+  file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h"
+                    "${CMAKE_CURRENT_SOURCE_DIR}/*.txx")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
     DESTINATION include/${PROJECT_NAME} COMPONENT Development)
-- 
2.54.0

